components/src/main/java/uk/gov/hmrc/components/organism/information/InformationMessageCardView.kt

```diff
@@ -92,26 +92,45 @@ class InformationMessageCardView @JvmOverloads constructor(
         }
     }
 
-    fun setHeadlineButtons(buttons: List<SecondaryButton>) {
+    fun setHeadlineButtons(buttons: List<InformationMessageButton>) {
         binding.headlineButtonsContainer.removeAllViews()
 
         val spacing8 = context.resources.getDimensionPixelSize(R.dimen.hmrc_spacing_8)
         val spacing16 = context.resources.getDimensionPixelSize(R.dimen.hmrc_spacing_16)
 
         binding.headlineButtonsContainer.visibility = if (buttons.isNotEmpty()) View.VISIBLE else View.GONE
         buttons.forEach {
-            it.minHeight = 0
-            it.setMargins(spacing16, spacing8, spacing16, 0)
-            it.backgroundTintList = ContextCompat.getColorStateList(context, R.color.hmrc_white)
-            it.gravity = Gravity.CENTER
-            binding.headlineButtonsContainer.addView(it)
+            it.button.apply {
+                minHeight = 0
+                setMargins(spacing16, spacing8, spacing16, 0)
+                gravity = Gravity.CENTER
+            }
+            if (it.isOutlineButton) {
+                it.button.apply {
+                    strokeWidth = OUTLINE_STROKE_WIDTH
+                    strokeColor = ContextCompat.getColorStateList(context, type!!.headlineTint)
+                    setTextColor(ContextCompat.getColorStateList(context, type!!.headlineTint))
+                }
+            } else {
+                it.button.backgroundTintList = ContextCompat.getColorStateList(context, R.color.hmrc_white)
+            }
+            binding.headlineButtonsContainer.addView(it.button)
         }
     }
 
+    data class InformationMessageButton(
```

Comment:
Do you think this would be tidier as a sealed class? Then rather than having to set true/false, you just define the one you want.

```
sealed class InformationMessageButton(val button: SecondaryButton, val isOutlineButton: Boolean {
  data class ActionButton(actionButton: Secondary button): InformationMessageButton(actionButton, false)
  data class OutlineButton(outlineButton: SecondaryButton): InformationMessageButton(outlineButton, true)
```